apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -22,46 +22,312 @@
+	}
+	free(comp_tc_data);
+	free(decomp_tc_data);
+}
+
+static int initialize_tc_data(void)
+{
+	comp_tc_data = (struct compress_header *)malloc(sizeof(struct compress_header));
+	if (comp_tc_data == NULL) {
+		return ERROR;
+	}
+		
+	comp_tc_data->input_buffer = (unsigned char *)malloc(INPUT_OUTPUT_SIZE);
+	if (comp_tc_data->input_buffer == NULL) {
+		goto exit_with_comp_input;
+	}
+
+	comp_tc_data->input_size = INPUT_OUTPUT_SIZE;
+	
+	for(int i = 0; i < comp_tc_data->input_size; i++) {
+		comp_tc_data->input_buffer[i] = 'a' + (rand() % 26);
+	}
+	
+	comp_tc_data->output_size = INPUT_OUTPUT_SIZE;
+	comp_tc_data->output_buffer = (unsigned char *)malloc(INPUT_OUTPUT_SIZE);
+	if (comp_tc_data->output_buffer == NULL) {
+		goto exit_with_comp_output;
+	}
+	
+	decomp_tc_data = (struct compress_header *)malloc(sizeof(struct compress_header));
+	if (decomp_tc_data == NULL) {
+		goto exit_with_decomp;
+	}
+		
+	decomp_tc_data->input_buffer = comp_tc_data->output_buffer;
+	decomp_tc_data->input_size = comp_tc_data->output_size;
+	
+	return OK;
+	
+exit_with_decomp:
+	free(comp_tc_data->output_buffer);
+exit_with_comp_output:
+	free(comp_tc_data->input_buffer);
+exit_with_comp_input:
+	free(comp_tc_data);
+	
+	return ERROR;
+}
+
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+static void tc_compressed_file_read(void)
+{
+	int ret_chk;
+	fd = tc2_get_drvfd();
+
+	/*If FAIL : Failed to open testcase driver*/
+	TC_ASSERT_GEQ("tc2_get_drvfd", fd, OK);
+
+	ret_chk = ioctl(fd, TESTIOC_COMPRESSION_TEST, 0);
+	TC_ASSERT_EQ("Compression_read", ret_chk, OK);
+
+	TC_SUCCESS_RESULT();
+}
 
 /****************************************************************************
- * Name: tc_compress_decompress
+ * Name: tc_comp_decom_buffer_known_output_size
  *
  * Description:
- *   Block-wise compression and decompression of a file.
+ *   Testing compression and decompression of a file when the
+ *   output size after decompression is known .
+ *   This is the case where output size is kept equal to 
+ *   actual size of decompressed data.
  *
  * Returned Value:
  *   None
  ****************************************************************************/
-static void tc_compress_decompress(void)
+static void tc_comp_decomp_buffer_known_output_size(void)
 {
-	int fd;
 	int ret_chk;
-	fd = tc2_get_drvfd();
+	fd = get_compfd();
 
 	/*If FAIL : Failed to open testcase driver*/
-	TC_ASSERT_GEQ("tc2_get_drvfd", fd, OK);
+	TC_ASSERT_GEQ("get_compfd", fd, OK);
+	
+	// initializing comp_tc_data
+	ret_chk = initialize_tc_data();
+	TC_ASSERT_EQ_CLEANUP("compression data initialize", ret_chk, OK, uninitialize_tc_data());
+	
+	// Test compression method 
+	ret_chk = ioctl(fd, COMPIOC_COMPRESS, comp_tc_data);
+	TC_ASSERT_EQ_CLEANUP("compress block", ret_chk, OK, uninitialize_tc_data());
+	
+	decomp_tc_data->input_size = comp_tc_data->output_size;
```

Comment:
I think we dont have to set this value. Its already set in the init function.

Reply:
Actually the comp_tc_data->output_size  changes after compression, thats why i need to assign the updated value.
Becuase in init function its alway INPUT_OUTPUT_SIZE(ex 10000) but after compression it changes to something less (ex 670), thats why i need to update.

Reply:
ok